src/OutputWriters/netcdf_writer.jl

```diff
@@ -760,669 +25,10 @@ mutable struct NetCDFWriter{G, D, O, T, A, FS, DN} <: AbstractOutputWriter
-                                    name,
-                                    array_type,
-                                    deflatelevel,
-                                    attributes,
-                                    dimensions,
-                                    filepath, # for better error messages
-                                    dimension_name_generator,
-                                    true) # time_dependent = true)
-        end
-
-        sync(dataset)
-    end
-
-    close(dataset)
-
-    return dataset, outputs, schedule
-end
-
-initialize_nc_file(ow::NetCDFWriter, model) = initialize_nc_file(model,
-                                                                 ow.filepath,
-                                                                 ow.outputs,
-                                                                 ow.schedule,
-                                                                 ow.array_type,
-                                                                 ow.indices,
-                                                                 ow.global_attributes,
-                                                                 ow.output_attributes,
-                                                                 ow.dimensions,
-                                                                 ow.with_halos,
-                                                                 ow.include_grid_metrics,
-                                                                 ow.overwrite_existing,
-                                                                 ow.deflatelevel,
-                                                                 ow.dimension_name_generator)
-
-#####
-##### Variable definition
-#####
-
-materialize_output(func, model) = func(model)
-materialize_output(field::AbstractField, model) = field
-materialize_output(particles::LagrangianParticles, model) = particles
-materialize_output(output::WindowedTimeAverage{<:AbstractField}, model) = output
-
-""" Defines empty variables for 'custom' user-supplied `output`. """
-function define_output_variable!(dataset, output, name, array_type,
-                                 deflatelevel, attrib, dimensions, filepath,
-                                 dimension_name_generator, time_dependent)
-
-    if name ∉ keys(dimensions)
-        msg = string("dimensions[$name] for output $name=$(typeof(output)) into $filepath" *
-                     " must be provided when constructing NetCDFWriter")
-        throw(ArgumentError(msg))
-    end
-
-    dims = dimensions[name]
-    FT = eltype(array_type)
-    defVar(dataset, name, FT, (dims..., "time"); deflatelevel, attrib)
-
+function NetCDFWriter(model, outputs; kw...)
+    msg = string("Building Oceananigans.OutputWriters.NetCDFWriter requires `using NCDatasets`.", '\n',
+                 "If you think you are already `using NCDatasets` , check for warnings or errors.")
+                      
+    error(msg)
```

Comment:
Again, I think this way to displaying the error message isn't ideal. I'd rather we rely on julia's `Method not found` error and just write `function NetCDFWriter end`.

Reply:
Are you talking about required kwargs? We can add those here

Reply:
Or do you mean a case where they pass an argument other than `model`? That's also easy to deal with. We can make this a `MethodError`

Reply:
```julia
julia> function h(args...; kw...)
           @warn "Did you load a package?"
           throw(MethodError(h, args))
           end

julia> h(1)
┌ Warning: Did you load a package?
└ @ Main REPL[12]:2
ERROR: MethodError: no method matching h(::Int64)

Closest candidates are:
  h(::Any...; kw...)
   @ Main REPL[12]:1

Stacktrace:
 [1] h(args::Int64; kw::@Kwargs{})
   @ Main ./REPL[12]:3
 [2] h(args::Int64)
   @ Main ./REPL[12]:1
 [3] top-level scope
   @ REPL[14]:1
```

Reply:
I'll mark this as solved. I was under the impression that your way of doing things masked the useful Julia messages if the user makes a mistake after `using NCDatasets`. But I confirmed that's not the case and the message is still useful. See my previous comment.

Reply:
I guess my only comment is that "relying on MethodError" and "writing `function f end` are not necessarily simultaneously required things. You can do one, or the other, or both.